src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
 
-    @Override
-    public void put(K key, V value) {
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
 
+        private Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
+
+    @Override
+    public V put(K key, V value) {
+        Node<K, V> valNode = getNode(key);
+        if (valNode != null) {
+            valNode.value = value;
+            return value;
+        }
+        if (size == threshold) {
+            resize();
+        }
+        addNode(getHashCode(key), key, value);
+        size++;
+        return null;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> neededNode = getNode(key);
+        while (neededNode != null) {
+            if (Objects.equals(key, neededNode.key)) {
+                return neededNode.value;
+            }
+            neededNode = neededNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public int getHashCode(K key) {
```

Comment:
The 'getHashCode' method should be private because it's only a utility method used within the class itself.